cookbooks/aws-parallelcluster-config/templates/default/init/fetch_and_run.erb

```diff
@@ -48,40 +48,34 @@ function get_stack_status () {
 
 function run_preinstall () {
   if [ "${cfn_preinstall}" != "NONE" ]; then
-    file="${cfn_preinstall}"
     if [ "${cfn_preinstall_args}" != "NONE" ]; then
-      download_run ${cfn_preinstall} "${cfn_preinstall_args[@]}"
+      download_run "${cfn_preinstall}" "${cfn_preinstall_args[@]}"
     else
-      download_run ${cfn_preinstall}
+      download_run "${cfn_preinstall}"
     fi
   fi || error_exit "Failed to run pre install hook"
 }
 
 function run_postinstall () {
   if [ "${cfn_postinstall}" != "NONE" ]; then
-    file="${cfn_postinstall}"
     if [ "${cfn_postinstall_args}" != "NONE" ]; then
-      download_run ${cfn_postinstall} "${cfn_postinstall_args[@]}"
+      download_run "${cfn_postinstall}" "${cfn_postinstall_args[@]}"
     else
-      download_run ${cfn_postinstall}
+      download_run "${cfn_postinstall}"
     fi
   fi || error_exit "Failed to run post install hook"
 }
 
 function run_postupdate () {
-  stack_status=$(get_stack_status) || error_exit "Failed to get the stack status"
-
-  if [ "${stack_status}" != "UPDATE_IN_PROGRESS" ]; then
-    echo "Post update hook called with CFN stack in state ${stack_status}, doing nothing"
-    exit 0
-  fi
-
   if [ "${cfn_postupdate}" != "NONE" ]; then
-    file="${cfn_postupdate}"
-    if [ "${cfn_postupdate_args}" != "NONE" ]; then
-      download_run ${cfn_postupdate} "${cfn_postupdate_args[@]}"
+    stack_status=$(get_stack_status) || error_exit "Failed to get the stack status, check the HeadNode instance profile's Iam policies"
```

Comment:
nitpick, IAM should probably be all capitals, [example in the docs](https://docs.aws.amazon.com/parallelcluster/latest/ug/HeadNode-v3.html#yaml-HeadNode-Iam-S3Access:~:text=Names%20(ARNs)%20of-,IAM,-policies%20for%20Amazon)